File: src/main/Java/boj_step11/boj24313.java
import java.io.*;
import java.util.StringTokenizer;

public class boj24313 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        long a1 = Long.parseLong(tokenizer.nextToken());
        long a0 = Long.parseLong(tokenizer.nextToken());
        long c = Long.parseLong(reader.readLine());
        long a = Long.parseLong(reader.readLine());
        writer.append((a1 * a + a0 <= c * a && c >= a1) ? "1" : "0").close();
    }
}

File: src/main/kotlin/boj_step11/boj24313.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val (a1, a0) = r.readLine().split(" ").map { it.toLong() }
    val c = r.readLine().toLong()
    val a = r.readLine().toLong()
    w.write((a1 * a + a0 <= c * a && c >= a1).compareTo(false).toString())
    w.close()
}